packages/loot-core/src/types/models/schedule.d.ts

```diff
@@ -31,3 +31,25 @@ export interface ScheduleEntity {
   _conditions: unknown;
   _actions: unknown;
 }
+
+export type DiscoverScheduleEntity = {
+  id: ScheduleEntity['id'];
+  account: AccountEntity['id'];
+  payee: PayeeEntity['id'];
+  date: ScheduleEntity['_date'];
+  amount: ScheduleEntity['_amount'];
+  _conditions: [
+    { op: 'is'; field: 'account'; value: AccountEntity['id'] },
+    { op: 'is'; field: 'payee'; value: PayeeEntity['id'] },
+    {
+      op: 'is' | 'isapprox';
+      field: 'date';
+      value: ScheduleEntity['_date'];
+    },
+    {
+      op: 'is' | 'isapprox';
+      field: 'amount';
+      value: ScheduleEntity['_amount'];
+    },
+  ];
```

Comment:
```suggestion
_conditions: Array<
    | { op: 'is'; field: 'account'; value: AccountEntity['id'] }
    | { op: 'is'; field: 'payee'; value: PayeeEntity['id'] }
    | {
        op: 'is' | 'isapprox';
        field: 'date';
        value: ScheduleEntity['_date'];
      }
    | {
        op: 'is' | 'isapprox';
        field: 'amount';
        value: ScheduleEntity['_amount'];
      }
  >;
```

Reply:
Thank you !

Reply:
You should use `Array` instead of `[]` here. I'm quite surprised it doesn't fail in CI..

But here's a playground so you can see the difference between the two types: https://www.typescriptlang.org/play?#code/C4TwDgpgBAKgFgSwM4CEBOECGBrJUC8UA3gFBTlQBumANgK4QBcUA2mRRwD7FQD2YzAOTJBAbigAzBBBoATIZgDGi3nQB2wMVAC+7DuW5E+AqMKRapM+abCYQECFt0cAuiV0lQkWIiQB1XjRcAmI9KloGZgBBNDQ7AB4wrh5+IRFxSzkFZVUNJySKQ2M08wzpLJs7B3yOAD53EhIVNSRgKGBfAH0AdwQaGh7A7GZ4ZACgvEJSDmp6JlYC8iNU0xEAGkly60Fbe0cdNcWUkzNBDcztpRV1TQOwtw9m1vau3v7OgCMMHBHfdCxglMwrNIgt9BRlid1psrEJdtU7uClsVVuZzltstc8oikZCSmcYRUdlV9tpDq53EA

Reply:
yeah !, My bad !!, Thank you :) 👍